[BUG] Email action: 'ConsumableDocument' object has no attribute 'correspondent'

### Description

To my existing workflow I added a new email action and entered all the required information.
The workflow triggers on 'Consumption Started' and there is a source filter 'consume folder'.
Now when I drop a new file into consume folder the file task goes into 'Failed' with the error 'ConsumableDocument' object has no attribute 'correspondent'

### Steps to reproduce

see above

### Webserver logs

```bash
[2025-01-15 10:58:31,115] [INFO] [paperless.parsing] convert exited 0
[2025-01-15 10:58:35,265] [INFO] [paperless.consumer] Document 2024-09-30 Kreuzweg consumption finished
[2025-01-15 10:58:35,279] [INFO] [paperless.tasks] ConsumeTaskPlugin completed with: Success. New document id 1924 created
[2025-01-15 10:58:35,296] [INFO] [celery.app.trace] Task documents.tasks.consume_file[c0192070-1b44-4f99-b75a-eba4ad208551] succeeded in 14.224035289138556s: 'Success. New document id 1924 created'
M2M reverse owner <ManyToOneRel: documents.document>
M2M reverse correspondent <ManyToOneRel: documents.document>
M2M reverse storage_path <ManyToOneRel: documents.document>
M2M reverse document_type <ManyToOneRel: documents.document>
M2M forward tags
[2025-01-15 10:58:57,143] [INFO] [paperless.tasks] Deleted 0 documents from trash
[2025-01-15 10:59:28,927] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/consume/Kreuzweg.pdf to the task queue.
[2025-01-15 10:59:29,040] [INFO] [celery.worker.strategy] Task documents.tasks.consume_file[0469545b-9c4d-46c9-b1ea-a4f3733834fb] received
[2025-01-15 10:59:29,325] [INFO] [paperless.matching] Document matched WorkflowTrigger 4 from Workflow: Besitzer zuweisen (Matthias)
[2025-01-15 10:59:29,335] [ERROR] [paperless.tasks] WorkflowTriggerPlugin failed: 'ConsumableDocument' object has no attribute 'correspondent'
Traceback (most recent call last):
  File "/usr/src/paperless/src/documents/tasks.py", line 154, in consume_file
    msg = plugin.run()
          ^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 64, in run
    overrides, msg = run_workflows(
                     ^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/signals/handlers.py", line 1116, in run_workflows
    email_action()
  File "/usr/src/paperless/src/documents/signals/handlers.py", line 918, in email_action
    correspondent = document.correspondent.name if document.correspondent else ""
                                                   ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConsumableDocument' object has no attribute 'correspondent'
[2025-01-15 10:59:29,430] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[0469545b-9c4d-46c9-b1ea-a4f3733834fb] raised unexpected: AttributeError("'ConsumableDocument' object has no attribute 'correspondent'")
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/celery/app/trace.py", line 453, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/tasks.py", line 154, in consume_file
    msg = plugin.run()
          ^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 64, in run
    overrides, msg = run_workflows(
                     ^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/signals/handlers.py", line 1116, in run_workflows
    email_action()
  File "/usr/src/paperless/src/documents/signals/handlers.py", line 918, in email_action
    correspondent = document.correspondent.name if document.correspondent else ""
                                                   ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConsumableDocument' object has no attribute 'correspondent'
```


### Browser logs

_No response_

### Paperless-ngx version

2.14.2

### Host OS

Debian 6.1.90-1

### Installation method

Docker - official image

### System status

_No response_

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/consumer.py
+++ b/src/documents/consumer.py
@@ -62,10 +62,10 @@ class WorkflowTriggerPlugin(
         Get overrides from matching workflows
         """
         overrides, msg = run_workflows(
-            WorkflowTrigger.WorkflowTriggerType.CONSUMPTION,
-            self.input_doc,
-            None,
-            DocumentMetadataOverrides(),
+            trigger_type=WorkflowTrigger.WorkflowTriggerType.CONSUMPTION,
+            document=self.input_doc,
+            logging_group=None,
+            overrides=DocumentMetadataOverrides(),
         )
         if overrides:
             self.metadata.update(overrides)

--- a/src/documents/signals/handlers.py
+++ b/src/documents/signals/handlers.py
@@ -11,6 +11,7 @@ from celery.signals import task_failure
 from celery.signals import task_postrun
 from celery.signals import task_prerun
 from django.conf import settings
+from django.contrib.auth.models import User
 from django.core.mail import EmailMessage
 from django.db import DatabaseError
 from django.db import close_old_connections
@@ -29,9 +30,11 @@ from documents.data_models import DocumentMetadataOverrides
 from documents.file_handling import create_source_path_directory
 from documents.file_handling import delete_empty_directories
 from documents.file_handling import generate_unique_filename
+from documents.models import Correspondent
 from documents.models import CustomField
 from documents.models import CustomFieldInstance
 from documents.models import Document
+from documents.models import DocumentType
 from documents.models import MatchingModel
 from documents.models import PaperlessTask
 from documents.models import Tag
@@ -907,21 +910,43 @@ def run_workflows(
             )
             return
 
-        title = (
-            document.title
-            if isinstance(document, Document)
-            else str(document.original_file)
-        )
-        doc_url = None
-        if isinstance(document, Document):
+        if not use_overrides:
+            title = document.title
             doc_url = f"{settings.PAPERLESS_URL}/documents/{document.pk}/"
-        correspondent = document.correspondent.name if document.correspondent else ""
-        document_type = document.document_type.name if document.document_type else ""
-        owner_username = document.owner.username if document.owner else ""
-        filename = document.original_filename or ""
-        current_filename = document.filename or ""
-        added = timezone.localtime(document.added)
-        created = timezone.localtime(document.created)
+            correspondent = (
+                document.correspondent.name if document.correspondent else ""
+            )
+            document_type = (
+                document.document_type.name if document.document_type else ""
+            )
+            owner_username = document.owner.username if document.owner else ""
+            filename = document.original_filename or ""
+            current_filename = document.filename or ""
+            added = timezone.localtime(document.added)
+            created = timezone.localtime(document.created)
+        else:
+            title = overrides.title if overrides.title else str(document.original_file)
+            doc_url = ""
+            correspondent = (
+                Correspondent.objects.filter(pk=overrides.correspondent_id).first()
+                if overrides.correspondent_id
+                else ""
+            )
+            document_type = (
+                DocumentType.objects.filter(pk=overrides.document_type_id).first().name
+                if overrides.document_type_id
+                else ""
+            )
+            owner_username = (
+                User.objects.filter(pk=overrides.owner_id).first().username
+                if overrides.owner_id
+                else ""
+            )
+            filename = document.original_file if document.original_file else ""
+            current_filename = filename
+            added = timezone.localtime(timezone.now())
+            created = timezone.localtime(overrides.created)
+
         subject = parse_w_workflow_placeholders(
             action.email.subject,
             correspondent,
@@ -973,21 +998,42 @@ def run_workflows(
             )
 
     def webhook_action():
-        title = (
-            document.title
-            if isinstance(document, Document)
-            else str(document.original_file)
-        )
-        doc_url = None
-        if isinstance(document, Document):
+        if not use_overrides:
+            title = document.title
             doc_url = f"{settings.PAPERLESS_URL}/documents/{document.pk}/"
-        correspondent = document.correspondent.name if document.correspondent else ""
-        document_type = document.document_type.name if document.document_type else ""
-        owner_username = document.owner.username if document.owner else ""
-        filename = document.original_filename or ""
-        current_filename = document.filename or ""
-        added = timezone.localtime(document.added)
-        created = timezone.localtime(document.created)
+            correspondent = (
+                document.correspondent.name if document.correspondent else ""
+            )
+            document_type = (
+                document.document_type.name if document.document_type else ""
+            )
+            owner_username = document.owner.username if document.owner else ""
+            filename = document.original_filename or ""
+            current_filename = document.filename or ""
+            added = timezone.localtime(document.added)
+            created = timezone.localtime(document.created)
+        else:
+            title = overrides.title if overrides.title else str(document.original_file)
+            doc_url = ""
+            correspondent = (
+                Correspondent.objects.filter(pk=overrides.correspondent_id).first()
+                if overrides.correspondent_id
+                else ""
+            )
+            document_type = (
+                DocumentType.objects.filter(pk=overrides.document_type_id).first().name
+                if overrides.document_type_id
+                else ""
+            )
+            owner_username = (
+                User.objects.filter(pk=overrides.owner_id).first().username
+                if overrides.owner_id
+                else ""
+            )
+            filename = document.original_file if document.original_file else ""
+            current_filename = filename
+            added = timezone.localtime(timezone.now())
+            created = timezone.localtime(overrides.created)
 
         try:
             data = {}

--- a/src/documents/tests/test_workflows.py
+++ b/src/documents/tests/test_workflows.py
@@ -2296,6 +2296,64 @@ class TestWorkflows(
         EMAIL_ENABLED=True,
         PAPERLESS_URL="http://localhost:8000",
     )
+    @mock.patch("httpx.post")
+    @mock.patch("django.core.mail.message.EmailMessage.send")
+    def test_workflow_email_consumption_started(self, mock_email_send, mock_post):
+        """
+        GIVEN:
+            - Workflow with email action and consumption trigger
+        WHEN:
+            - Document is consumed
+        THEN:
+            - Email is sent
+        """
+        mock_post.return_value = mock.Mock(
+            status_code=200,
+            json=mock.Mock(return_value={"status": "ok"}),
+        )
+        mock_email_send.return_value = 1
+
+        trigger = WorkflowTrigger.objects.create(
+            type=WorkflowTrigger.WorkflowTriggerType.CONSUMPTION,
+        )
+        email_action = WorkflowActionEmail.objects.create(
+            subject="Test Notification: {doc_title}",
+            body="Test message: {doc_url}",
+            to="user@example.com",
+            include_document=False,
+        )
+        action = WorkflowAction.objects.create(
+            type=WorkflowAction.WorkflowActionType.EMAIL,
+            email=email_action,
+        )
+        w = Workflow.objects.create(
+            name="Workflow 1",
+            order=0,
+        )
+        w.triggers.add(trigger)
+        w.actions.add(action)
+        w.save()
+
+        test_file = shutil.copy(
+            self.SAMPLE_DIR / "simple.pdf",
+            self.dirs.scratch_dir / "simple.pdf",
+        )
+
+        with mock.patch("documents.tasks.ProgressManager", DummyProgressManager):
+            with self.assertLogs("paperless.matching", level="INFO"):
+                tasks.consume_file(
+                    ConsumableDocument(
+                        source=DocumentSource.ConsumeFolder,
+                        original_file=test_file,
+                    ),
+                    None,
+                )
+
+        mock_email_send.assert_called_once()
+
+    @override_settings(
+        PAPERLESS_URL="http://localhost:8000",
+    )
     @mock.patch("documents.signals.handlers.send_webhook.delay")
     def test_workflow_webhook_action_body(self, mock_post):
         """
@@ -2352,8 +2410,6 @@ class TestWorkflows(
         )
 
     @override_settings(
-        PAPERLESS_EMAIL_HOST="localhost",
-        EMAIL_ENABLED=True,
         PAPERLESS_URL="http://localhost:8000",
     )
     @mock.patch("documents.signals.handlers.send_webhook.delay")
@@ -2415,8 +2471,6 @@ class TestWorkflows(
         )
 
     @override_settings(
-        PAPERLESS_EMAIL_HOST="localhost",
-        EMAIL_ENABLED=True,
         PAPERLESS_URL="http://localhost:8000",
     )
     def test_workflow_webhook_action_fail(self):
@@ -2562,3 +2616,56 @@ class TestWorkflows(
                     "Failed attempt sending webhook to http://paperless-ngx.com"
                 )
                 self.assertIn(expected_str, cm.output[0])
+
+    @mock.patch("documents.signals.handlers.send_webhook.delay")
+    def test_workflow_webhook_action_consumption(self, mock_post):
+        """
+        GIVEN:
+            - Workflow with webhook action and consumption trigger
+        WHEN:
+            - Document is consumed
+        THEN:
+            - Webhook is sent
+        """
+        mock_post.return_value = mock.Mock(
+            status_code=200,
+            json=mock.Mock(return_value={"status": "ok"}),
+        )
+
+        trigger = WorkflowTrigger.objects.create(
+            type=WorkflowTrigger.WorkflowTriggerType.CONSUMPTION,
+        )
+        webhook_action = WorkflowActionWebhook.objects.create(
+            use_params=False,
+            body="Test message: {doc_url}",
+            url="http://paperless-ngx.com",
+            include_document=False,
+        )
+        action = WorkflowAction.objects.create(
+            type=WorkflowAction.WorkflowActionType.WEBHOOK,
+            webhook=webhook_action,
+        )
+        w = Workflow.objects.create(
+            name="Workflow 1",
+            order=0,
+        )
+        w.triggers.add(trigger)
+        w.actions.add(action)
+        w.save()
+
+        test_file = shutil.copy(
+            self.SAMPLE_DIR / "simple.pdf",
+            self.dirs.scratch_dir / "simple.pdf",
+        )
+
+        with mock.patch("documents.tasks.ProgressManager", DummyProgressManager):
+            with self.assertLogs("paperless.matching", level="INFO"):
+                tasks.consume_file(
+                    ConsumableDocument(
+                        source=DocumentSource.ConsumeFolder,
+                        original_file=test_file,
+                    ),
+                    None,
+                )
+
+        mock_post.assert_called_once()

